File: leetcode/55.jump-game/ts/canjump.ts
const canJump = (nums: number[]): boolean => {
    let maximum = 0

    for (let i = 0; i < nums.length; i++) {
        if (maximum >= nums.length - 1) return true

        if (i > maximum) return false

        if (i + nums[i] > maximum) maximum = i + nums[i]
    }

    return false
}

File: leetcode/55.jump-game/go/canjump.go
package jumpgame

func canJump(nums []int) bool {
	maximum := 0

	for i, v := range nums {
		if maximum >= len(nums) - 1 {
			return true
		}

		if i > maximum { // cannot access
			return false
		}

		if i + v > maximum {
			maximum = i + v
		}
	}

	return false
}
